### PR TITLE
Update LM Generalists

### PR DESCRIPTION
**Pull request by anwai98**

### CODE REVIEW

File: finetuning/specialists/training/light_microscopy/common.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import numpy as np
+from math import ceil, floor
+
+from torch_em.transform.label import label_consecutive
+from torch_em.transform.label import PerObjectDistanceTransform
+from torch_em.transform.raw import normalize_percentile, normalize
+
+
+class ResizeRawTrafo:
```
Comment (constantinpape) [inline finetuning/specialists/training/light_microscopy/common.py:9]: Let's not duplicate this functionality over and over. It's better if you move this to `micro_sam.training.util`.
Same is true for `ResizeLabelTransform`. And you should also do this for the code in https://github.com/computational-cell-analytics/micro-sam/pull/295.

File: finetuning/specialists/training/light_microscopy/deepbacs_finetuning.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    model.to(device)
+
+    # let's get the UNETR model for automatic instance segmentation pipeline
+    unetr = UNETR(
+        backbone="sam",
+        encoder=model.sam.image_encoder,
+        out_channels=3,
+        use_sam_stats=True,
+        final_activation="Sigmoid",
+        use_skip_connection=False
```
Comment (constantinpape) [inline finetuning/specialists/training/light_microscopy/deepbacs_finetuning.py:76]: `resize_input=True`

File: finetuning/specialists/training/light_microscopy/common.py (hunk 1 above)
Comment (anwai98) [reply finetuning/specialists/training/light_microscopy/common.py:9]: That's a good point. I will refactor these later then (once the joint training bug is taken care of). Thanks!

File: finetuning/generalists/training/light_microscopy/train_lm_generalist.py [hunk 3]
```diff
@@ -23,19 +25,43 @@ def finetune_lm_generalist(args):
...
+    model.to(device)
+
+    # let's get the UNETR model for automatic instance segmentation pipeline
+    unetr = UNETR(
+        backbone="sam",
+        encoder=model.sam.image_encoder,
+        out_channels=3,
+        use_sam_stats=True,
+        final_activation="Sigmoid",
+        use_skip_connection=False
```
[the remark this answers is not in the document]
Comment (anwai98) [reply finetuning/generalists/training/light_microscopy/train_lm_generalist.py:43]: The current default takes care of it. I would still update it likewise. Thanks!

File: finetuning/specialists/training/light_microscopy/common.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        return labels
+
+
+def deepbacs_raw_trafo(raw):
+    raw = normalize(raw)
+    raw = raw * 255
+    return raw
+
+
+def deepbacs_label_trafo(labels):
```
[the remark this answers is not in the document]
Comment (anwai98) [reply finetuning/specialists/training/light_microscopy/common.py:63]: Yes, `PerObjectDistanceTransform` takes care of all the components. It's not needed here. Thanks for pointing it out!

**Comment by anwai98** [comment]
This looks good to go now! The raw and label trafo functions for the specialist and generalist finetuning have been updated here. Once this is merged, I will update the experiments for https://github.com/computational-cell-analytics/micro-sam/pull/295 likewise. Thanks.

**Review by constantinpape — COMMENTED** [review]
Looks good now.  But I think we should adapt to changes from #301 before merging.

**Comment by anwai98** [comment]
This is good to go now. Thanks!

**Review by constantinpape — APPROVED** [review]

**Pull request merged**